CBDatabase/Protocols/DatabaseModelObject.swift

```diff
@@ -12,7 +12,30 @@ import CoreData
 ///         - Add the entity's attributes as ivars.
 ///         - If the entity name differs from the name of the struct, be sure to override entityName.
 public protocol DatabaseModelObject: Codable, Hashable {
+    static var entityName: String { get }
+}
+
+public protocol IdentifiableDatabaseModelObject: DatabaseModelObject {
     var id: String { get }
+    static var idField: String { get }
```

Comment:
To keep things simple, I would say let's not allow developers to customize the `idField` name. i.e. always use `id` as the column name. If the developer wants to use anything custom, use `DatabaseModelObject` protocol instead.

Reply:
Hmm ok. All our ID's are like 'userID' but we can just always use DatabaseModelObject.

Reply:
We do have couple options here:
1. Since Consumer is migrating to this framework, you could just migrate the `userID` to default `id`
2. Or just create own custom protocol on Consumer side that extends `DatabaseModelObject` and adds custom `idField`